assets/src/components/app.tsx

```diff
@@ -96,6 +97,10 @@ export const AppRoutes = () => {
 }
 
 const App = (): ReactElement<HTMLDivElement> => {
+  useEffect(() => {
+    document.documentElement.dataset.demoMode =
+      inTestGroup("demo-mode").toString()
+  })
```

Comment:
**question:** does this need to be in a `useEffect`? If so, should it have an empty dependency array since the return value of `inTestGroup` will only change on a page reload? Although maybe there's something I'm overlooking about `documentElement` here.

Reply:
Thanks for catching this! I had meant for it to be a empty dependency array, I kidna thought it would do that by default and forgot to check my knowledge against the docs.


I don't know if a `useEffect` is the right tool here, I effectively wanted a way to run code once when we create the application and figured this was a good place to start. If you can think of a better place to put this I'm all for it.

Reply:
I tried to address this in 068fd33267fc1302ace44398b07aa7c932bb58ac, I'm curious to see what you think!

Reply:
I think that makes sense! Unfortunately we don't have tests for that `app.tsx` but we do have [an Asana ticket](https://app.asana.com/0/1152340551558956/1202219847571509/f) to add some.